### DISCUSSION TITLE
Different search results from faiss both with flat L

### DISCUSSION THREAD

**Discussion by jmugan** [milvus-io/milvus#16768] [Q&A and General discussion] [answered]
I'm using Milvus with an index of `{"index_type": "FLAT", "metric_type": 'L2',"params":{}}` and I'm getting different search results compared with the same data directly in faiss with `faiss.IndexFlatL2`. 

Since they are both exhaustive and both L2, they should be exactly the same, no? The milvus vectors seem to be farther away overall from the query vectors. 

The results are similar but not the same, with Milvus being a little worse. Anyone have any idea of what might be happening or what I might be doing wrong?

Thanks

**Comment by xiaofan-luan** [comment]
umm, I believe FLAT should be 100% accurate so milvus and FAISS should return exactly same result.
How large is your dataset? If it can be reproduced on a small dataset maybe we can help on deep investigation, thanks @jmugan

**Reply by jmugan** [reply]
That's what I would expect, and they are *almost* the same. I was hoping someone could say something along the lines of a parameter I need to set or something. I'll look at creating a minimal example.

**Reply by xiaofan-luan** [reply]
> That's what I would expect, and they are _almost_ the same. I was hoping someone could say something along the lines of a parameter I need to set or something. I'll look at creating a minimal example.

Cool, let us know if it's reproducible, I guess it might be some floating point precision issue in Go

**Reply by jmugan** [reply]
Will do. It's looking less and less like a Milvus issue. I'll keep you informed. Thanks!

**Reply by jmugan** [reply]
Figured it out. I had to flush. This is a pretty big gotcha since it isn't in the examples for new users.

I had to do
```
conn = collection._get_connection()
conn.flush([collection._name])
```
I'll make this an issue since for newcomers it's a pretty subtle way to get tripped up.

**Comment by jmugan** [comment, marked as answer]
I had to flush
```
conn = collection._get_connection()
conn.flush([collection._name])
```
